axon-5/api-changes.md

```diff
@@ -117,6 +117,12 @@ The fields can respectively be used to define the `Class#getSimpleName`, the pac
 `Message` it is connected too.
 This layer of indirection will allow us to provide the freedom that currently is not an option (as explained above).
 
+### Factory Methods, like GenericMessage#asMessage(Object)
+
+The factory methods that would construct a `Mesage` implementation based on a given `Object` have been removed from Axon Framework.
+These factory methods no longer align with the new API, which expects that the `QualifiedName` is set consciously.
+Hence, users of the factory methods need to revert to using the constructor of the `Message` implementation instead.
```

Comment:
This was an initial idea, but I'm not sure if we should follow that approach. 
First of all we need to find some place where to keep logic like in `GenericEventMessage` with handling case when payload is already a Message:
```
@Deprecated
    @SuppressWarnings("unchecked")
    public static <P> EventMessage<P> asEventMessage(@Nonnull Object event) {
        if (event instanceof EventMessage) {
            return (EventMessage<P>) event;
        } else if (event instanceof Message) {
            Message<P> message = (Message<P>) event;
            return new GenericEventMessage<>(message, clock.instant());
        }
        return new GenericEventMessage<>(
                new GenericMessage<>(/*resolving QualifiedName*/, (P) event),
                clock.instant()
        );
    }
```

Reply:
Decision: drop support for message as a payload in Axon Framework 5.